index.bs

```diff
@@ -552,13 +552,52 @@ spec: XHR; urlPrefix: https://xhr.spec.whatwg.org/
 
   Cross-origin messaging via <a>`postMessage`</a> requires that the recipient be
   able to see the suborigin namespace of the message sender so it can make an
-  appropriate access control decision. When a message is sent from a suborigin
-  namespace, the receiver has the `event.origin` value set to the serialized
-  suborigin value instead of the serializied origin, as described in
-  [[#serializing]]. Additionally, a new `suborigin` property must be added to
-  the `MessageEvent` given to the receiver which contains the suborigin
-  namespace value.
+  appropriate access control decision. We introduce a new dictionary
+  `EventInitiator` that holds information about both the origin and
+  the suborigin namespace:
+
+  <pre class="idl">
+    dictionary EventInitiator {
+      USVString origin = "";
+      USVString? suborigin = null;
+    };
+  </pre>
+
+  And we extend the definition of the MessageEvent and MessageEventInit as follows:
+
+  <pre class="idl">
+    partial interface MessageEvent {
+      USVString? origin;
+      EventInitiator? initiator;
+      void initMessageEvent(DOMString type, optional boolean bubbles = false, optional boolean cancelable = false, optional any data = null, optional EventInitiator initator? = null, optional DOMString lastEventId = "", optional MessageEventSource? source = null, optional sequence&lt;MessagePort&gt; ports = []);
+    };
+
+    partial interface MessageEventInit : EventInit {
+      EventInitiator? initiator = null;
+    };
+  </pre>
+  
+  Similarly, we add a new overloaded version of postMessage to the Window interface:
+
+  <pre class="idl">
+    partial interface Window {
+      void postMessage(any message, EventInitiator targetOrigin, optional sequence&lt;object&gt; transfer = []);
+    };
+  </pre>
 
+  Those new interfaces can also be used to communicate with browsing contexts that
+  aren't in a suborigin namespace. Furthermore, the `origin` attribute of an `MessageEvent`
+  will be `null` (the value, not the string) if the `suborigin` attribute of
+  the event's `initiator` is non-`null`, unless `unsafe-postmessage-receive` was specified
```

Comment:
Right, but does that make sense? If you assume that the app inside the suborigin doesn't even support suborigins, how likely is it that the other side supports them? So I'd expect that in your model, you will always need both options.

My model is that you can say "the app in the suborigin actually understands what it is doing" and "the app outside of the suborigin understands what it is doing".